engine/vcs/bitbucketserver/http.go

```diff
@@ -191,11 +191,75 @@ func (c *bitbucketClient) do(ctx context.Context, method, api, path string, para
 			}
 		}
 		if method == "GET" {
-			if err := c.consumer.cache.Set(cacheKey, v); err != nil {
+			if err := b.consumer.cache.Set(cacheKey, v); err != nil {
 				log.Error(ctx, "unable to cache set %v: %v", cacheKey, err)
 			}
 		}
 	}
 
 	return nil
 }
+
+func (b *bitbucketClient) stream(ctx context.Context, method, api, path string, params url.Values, values []byte) (io.Reader, http.Header, error) {
```

Comment:
we should consider using a lib for the bitbucket client....

Reply:
yes